main.ts

```diff
@@ -164,10 +220,44 @@ export default class ShareAsGistPlugin extends Plugin {
       }),
     });
 
+    this.addTimeoutUpdater();
+
     // This adds a settings tab so the user can configure various aspects of the plugin
     this.addSettingTab(new ShareAsGistSettingTab(this.app, this));
   }
 
+  addTimeoutUpdater() {
+    let timeout: number | null = null;
+    let previousContent = '';
+
+    this.app.workspace.on('editor-change', async (editor, _info) => {
```

Comment:
I'm wondering if the `editor-change` event is the best thing to hook into here. Would it make more sense to hook into when the file is saved instead? It looks like you can do that with the `modify` event - although I haven't researched that in detail:

https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts#L4156

Reply:
I don't know how I didn't see that, I'll test it out when I get a chance

Reply:
LGTM